c/graphLib/io/g6-write-iterator.c

```diff
@@ -461,6 +467,12 @@ int _printEncodedGraph(G6WriteIterator *pG6WriteIterator)
 		exitCode = NOTOK;
 	}
 
+	if (sf_fputs("\n", pG6WriteIterator->g6Output) < 0)
```

Comment:
This is a good add. One ask is if you could run an experiment on Windows to see if instead using "\r\n" will write "\r\n" (correct) or "\r\r\n" (incorrect). If the output is correct ("\r\n" produces "\r\n" in the file output), then it would be better to amend this PR as follows:
1. add a LINE_DELIMITER macro in appconst.h just below FILE_DELIMITER and set equal to "\r\n" for WINDOWS and "\n" otherwise.
2. Use LINE_DELIMITER in this call instead of just "\n"

The advantage of this approach, if the output is correct, is that it will write "\r\n" when you are writing to string on Windows, and not just when writing to file.

Reply:
When I wrote

```
sf_fputs("\r\n", pG6WriteIterator->g6Output)
```
when `pG6WriteIterator->g6Output` contains a `FILE *`, I observed that the resulting `.g6` output file converted both characters to `CRLF` (i.e. two `CRLF between each graph).

When writing strings *only* (i.e. if writing to file, then allow `fputs()` to continue doing it's thing), should I try to replace any line terminator within the string (e.g. `\r`, `\n`, or `\r\n`) with the `LINE_DELIMITER` character that's based on whether `#ifdef WINDOWS`, or should I leave as-is? 🎉

Reply:
Ah, ok, thanks for checking. I've been reluctant for many years to add this LINE_DELIMITER business, and even though it's disappointing that `fputs()` did what I consider to be the wrong thing (writing two CRLFs in response to having one CRLF in the string), it is kind of a relief to not have to add LINE_DELIMITER now. The right answer in your `sf_fputs()` call is to use `"\n"`, as you have done, because the methods we're calling (like the real `fputs()` and `printf()`) are turning `\n` into a CRLF on windows, as we need them to do. 

It's a good question whether the string side of `sf_fputs()` should be converting `\n` to CRLF. The right answer is to leave it as is, which seems counterintuitive from the standpoint of wanting to be consistent with methods of the language, like `fputs()` and `printf()`. However, we're still on the outside of those language methods, so if we think of how the strings we write will get used, they will go into those language methods, which will then turn our CRLFs into double CRLFs. 

So now the only question is how I can turn off the "Change requested" flag on this PR because I should have just submitted it as a review comment without requesting a change yet, since the ask was for an experiment first, and the net result of that experiment is that we should merge this PR without a change.